llvm/include/llvm/ABI/Types.h

```diff
@@ -0,0 +1,453 @@
+//===- ABI/Types.h ----------------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file defines the Types and related helper methods concerned to the
+/// LLVMABI library which mirrors ABI related type information from
+/// the LLVM frontend.
```

Comment:
This whole comment reads a bit strangely, e.g. "concerned to" is not normal English phrasing. Maybe rewrite to something like:

> The file defines the type system of the LLVMABI library, which frontends can use to describe the aspects of their high level types that affect the ABI.

(I'm sure this can still be improved!)